python/lsst/daf/persistence/posixStorage.py

```diff
@@ -348,5 +417,159 @@ def v1RepoExists(root):
         return os.path.exists(root) and bool(os.listdir(root))
 
 
+    ####################################
+    # PosixStorage-only API (for now...)
+
+    def copyFile(self, fromLocation, toLocation):
+        shutil.copy(os.path.join(self.root, fromLocation), os.path.join(self.root, toLocation))
+
+    def getRoot(self):
+        return self.root
+
+    def getLocalFile(self, path):
+        """Get the path to a local copy of the file, downloading it to a temporary if needed.
+
+        Parameters
+        ----------
+        A path the the file in storage, relative to root.
+
+        Returns
+        -------
+        A path to a local copy of the file. May be the original file (if storage is local)."""
+        p = os.path.join(self.root, path)
+        if os.path.exists(p):
+            return p
+        else:
+            return None
+
+    def instanceSearch(self, path):
+        """Search for the given path in this storage instance.
+
+        If the path contains an HDU indicator (a number in brackets before the
+        dot, e.g. 'foo.fits[1]', this will be stripped when searching and so
+        will match filenames without the HDU indicator, e.g. 'foo.fits'. The
+        path returned WILL contain the indicator though, e.g. ['foo.fits[1]'].
+
+        Parameters
+        ----------
+        path : string
+            A filename (and optionally prefix path) to search for within root.
+
+        Returns
+        -------
+        string or None
+            The location that was found, or None if no location was found.
```

Comment:
I don't actually know, it's cargo culted to here from obs_base.CameraMapper. I tried to see where it originated but I can't figure out how to git blame it from before the change from daf_butlerUtils to obs_base. I'd like to leave it as is for now.